### PR TITLE
Add listing contract, add test, restructure slightly

### PR DESCRIPTION
**Pull request by samuelWilliams99**

### CODE REVIEW

File: src/Seabug/Contract/MarketPlaceSell.purs [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
+import Seabug.Types (MarketplaceDatum(..))
+
+-- | Mint the self-governed NFT for the given collection.
+marketPlaceSell
+  :: forall (r :: Row Type)
+   . CurrencySymbol /\ TokenName
+  -> Contract r TransactionHash
+marketPlaceSell (curr /\ tn) = do
+  owner <- liftedM "Cannot get PaymentPubKeyHash" ownPaymentPubKeyHash
+  ownerStake <- liftedM "Cannot get StakePubKeyHash" ownStakePubKeyHash
```
Comment (t4ccer) [inline src/Seabug/Contract/MarketPlaceSell.purs:38]: Should it throw error when there is no staking key? Maybe constructing an address without staking key would be more desired

File: src/Seabug/Contract/Util.purs [hunk 2]
```diff
@@ -78,6 +80,20 @@ modify fn t = wrap (fn (unwrap t))
...
+  -> (Value.CurrencySymbol /\ Value.TokenName)
+  -> TxConstraints Void Void
+payBehaviour ToCaller _ _ = mempty -- Balancing will return the token to the caller
+payBehaviour ToMarketPlace valHash asset =
+  mustPayToScript
+    valHash
+    ( Datum $ toData $
+        MarketplaceDatum { getMarketplaceDatum: asset }
+    )
+    (Value.singleton (fst asset) (snd asset) one)
```
Comment (t4ccer) [inline src/Seabug/Contract/Util.purs:95]: How does balancing/min ADA requirement work? Will some balancing mechanism add missing ADA here?

Comment (samuelWilliams99) [reply src/Seabug/Contract/Util.purs:95]: yes, it does, as the plutip tests pass, but I'll add it in explicit regardless

File: src/Seabug/Contract/MarketPlaceSell.purs (hunk 1 above)
Comment (samuelWilliams99) [reply src/Seabug/Contract/MarketPlaceSell.purs:38]: Good point, copied this from the minting contract, ill update it

**Pull request merged**